test/testdata/infer/flatten.rb

```diff
@@ -22,6 +22,26 @@
   T::Array[T::Array[T::Array[Integer]]],
 )
 
+class ToAry
+  extend T::Sig
+
+  sig { params(left: Integer, right: Integer).void }
+  def initialize(left, right)
+    @left = left
+    @right = right
+  end
+
+  sig { returns(T::Array[Integer]) }
+  def to_ary
+    [@left, @right]
+  end
+end
+
+flat_to_ary = T.let(
+  [ToAry.new(1, 2), ToAry.new(3, 4)],
+  T::Array[ToAry]
+)
```

Comment:
Thank you for adding a test.  Can you also add a test where `to_ary` is not defined on the class, but on a superclass?  I believe your current implementation will fail in such a case because the `findMember` check you used above will only check in the current class for such a symbol.